### PR TITLE
Fixes Nar'Sie rune erasing bug

### LINKED ISSUE #54413: Nar'Sie Rune is Not Clearable With Null Rod
image

At the least even if this is somehow intended behaviour the fact it's logged as removal in admin logs is a bug then.
Kinda bummed out rushing into a cult summon and spamming the rune with a null rod only to have nothing happen I recall nar'sie rune being clearable in the past.
Other runes don't have this issue.

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request

Gets rid of the nar'sie summoning rune's attackby overwrite and uses generic vars on the runes in the base rune's attackby to handle that behavior, also gives a confirmation to prevent accidentally erasing any important runes.
fixes: https://github.com/tgstation/tgstation/issues/54413

## Why It's Good For The Game

It was clearly intended to allow you to remove them, but it wasn't doing it.
No more heroic dives into mobs of cultists in an attempt to erase the rune last second just to find you can't any more.

## Changelog
:cl:
fix: Nar'Sie summoning runes can be erased with a null rod, or by a cultist with a sacrificial dagger after a confirmation message.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/cult/runes.dm [hunk 1]
```diff
@@ -57,11 +59,21 @@ Runes can either be invoked by one's self or with many different cultists. Each
 
 /obj/effect/rune/attackby(obj/I, mob/user, params)
 	if(istype(I, /obj/item/melee/cultblade/dagger) && iscultist(user))
+		if(log_when_erased)
+			var/confirm = alert(user, "Erasing this [src.cultist_name] rune might be against your goal to summon Nar'Sie.", "Begin to erase the [src.cultist_name] rune?", "Proceed", "Abort")
```
Comment (Jared-Fogle) [inline code/modules/antagonists/cult/runes.dm:63]: This and everywhere else, change `src.cultist_name` to `cultist_name`--src is implied.

File: code/modules/antagonists/cult/runes.dm [hunk 2]
```diff
@@ -29,6 +29,8 @@ Runes can either be invoked by one's self or with many different cultists. Each
 	var/req_cultists = 1 //The amount of cultists required around the rune to invoke it. If only 1, any cultist can invoke it.
 	var/req_cultists_text //if we have a description override for required cultists to invoke
 	var/rune_in_use = FALSE // Used for some runes, this is for when you want a rune to not be usable when in use.
+	var/log_when_erased = FALSE //Used when you want to keep track of who erased the rune
+	var/erase_time = 15 //How the rune takes to erase
```
Comment (Jared-Fogle) [inline code/modules/antagonists/cult/runes.dm:33]:
```suggestion
	var/erase_time = 1.5 SECONDS //How the rune takes to erase
```

File: code/modules/antagonists/cult/runes.dm [hunk 3]
```diff
@@ -462,6 +474,8 @@ structure_check() searches for nearby cultist structures required for the invoca
 	pixel_y = -32
 	scribe_delay = 500 //how long the rune takes to create
 	scribe_damage = 40.1 //how much damage you take doing it
+	log_when_erased = TRUE
+	erase_time = 50
```
Comment (Jared-Fogle) [inline code/modules/antagonists/cult/runes.dm:478]:
```suggestion
	erase_time = 5 SECONDS
```

**Comment by Jared-Fogle** [comment]
After those last nitpicks, LGTM.

File: code/modules/antagonists/cult/runes.dm [hunk 4]
```diff
@@ -57,11 +59,21 @@ Runes can either be invoked by one's self or with many different cultists. Each
 
 /obj/effect/rune/attackby(obj/I, mob/user, params)
 	if(istype(I, /obj/item/melee/cultblade/dagger) && iscultist(user))
+		if(log_when_erased)
+			var/confirm = alert(user, "Erasing this [cultist_name] rune might be against your goal to summon Nar'Sie.", "Begin to erase the [cultist_name] rune?", "Proceed", "Abort")
```
Comment (ATH1909) [inline code/modules/antagonists/cult/runes.dm:63]: input stalling is possible with this, if you for some reason want to erase your own summoning rune as a cultist without having a knife in your active hand

Comment (Jared-Fogle) [reply code/modules/antagonists/cult/runes.dm:63]: Fair point. It's not realistic, but you should check that the user can still reasonably do this after the input is finished (has dagger in active hand, is cultist, and is alive/not incapacitated/whatever).

Comment (ArcaneDefence) [reply code/modules/antagonists/cult/runes.dm:63]: I'd just ban you twice as hard for input stalling to grief your team anyway.

Comment (Jared-Fogle) [reply code/modules/antagonists/cult/runes.dm:63]: Yeah, but this is still something that should be solved through code.

**Review by Mothblocks — DISMISSED** [review]

File: code/modules/antagonists/cult/runes.dm [hunk 5]
```diff
@@ -57,11 +59,24 @@ Runes can either be invoked by one's self or with many different cultists. Each
...
+			if(confirm == "Abort")
+				return
+		var/held_item = user.get_active_held_item()
+		if(!(istype(held_item, /obj/item/melee/cultblade/dagger)) || !Adjacent(user) || user.incapacitated() || user.stat == DEAD) //Gee, good thing we made sure cultists can't input stall to grief their team and get banned anyway
```
Comment (ShizCalev) [inline code/modules/antagonists/cult/runes.dm:67]: user.is_holding_item_of_type(/obj/item/melee/cultblade/dagger)

File: code/modules/antagonists/cult/runes.dm [hunk 6]
```diff
@@ -57,11 +59,24 @@ Runes can either be invoked by one's self or with many different cultists. Each
 
 /obj/effect/rune/attackby(obj/I, mob/user, params)
 	if(istype(I, /obj/item/melee/cultblade/dagger) && iscultist(user))
+		if(log_when_erased)
+			var/confirm = alert(user, "Erasing this [cultist_name] rune might be against your goal to summon Nar'Sie.", "Begin to erase the [cultist_name] rune?", "Proceed", "Abort")
+			if(confirm == "Abort")
```
Comment (ShizCalev) [inline code/modules/antagonists/cult/runes.dm:64]: =! "Proceed"

to handle in case they close the window without clicking either button

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/cult/runes.dm [hunk 7]
```diff
@@ -57,11 +59,23 @@ Runes can either be invoked by one's self or with many different cultists. Each
 
 /obj/effect/rune/attackby(obj/I, mob/user, params)
 	if(istype(I, /obj/item/melee/cultblade/dagger) && iscultist(user))
+		if(log_when_erased)
+			var/confirm = alert(user, "Erasing this [cultist_name] rune might be against your goal to summon Nar'Sie.", "Begin to erase the [cultist_name] rune?", "Proceed", "Abort")
```
Comment (ShizCalev) [inline code/modules/antagonists/cult/runes.dm:63]: will this alert show up if someone holding a dagger clicks it with TK?

Comment (Jared-Fogle) [reply code/modules/antagonists/cult/runes.dm:63]: Don't you need an empty hand for TK?

Comment (ArcaneDefence) [reply code/modules/antagonists/cult/runes.dm:63]: It does give you the alert.

**Pull request merged**